cmd/deploy/remote.go

```diff
@@ -80,7 +86,7 @@ RUN okteto registrytoken install --force --log-output=json
 
 RUN --mount=type=secret,id=known_hosts --mount=id=remote,type=ssh \
   mkdir -p $HOME/.ssh && echo "UserKnownHostsFile=/run/secrets/known_hosts" >> $HOME/.ssh/config && \
-  okteto deploy --log-output=json --server-name="${{ .InternalServerName }}" {{ .DeployFlags }}
+  /okteto/bin/okteto remote-run --log-output=json --server-name="${{ .InternalServerName }}" {{ .CommandFlags }}
```

Comment:
This change is the one I mentioned in the description. I'm changing the command to use the whole path of the binary we get from our image to prevent a breaking change for those who might be using a custom image with a pinned CLI version (we are setting the CLI with the least priority in the PATH).